app/models/metadatum.rb

```diff
@@ -154,10 +163,36 @@ def check_and_set_location_type
       location.save!
     end
 
+    return {
+      location_id: location.id,
+      string_validated_value: nil,
+      raw_value: nil,
+    }
+  end
+
+  def check_and_set_location_type
+    # Skip if location was already resolved
+    return if location_id && !raw_value
+
+    # Increment this if you ever change the behavior of _determine_location_params or underlying functions.
+    cache_version_key = 1
+    cache_key = "#{raw_value}&_v=#{cache_version_key}"
+
+    # Use a cache for this function so that, if the exact same raw location value is passed for multiple samples
+    # in one request, we only validate it once, which saves time.
+    # Use the raw value as the cache key.
+    # Use a very short cache expiration time because our current performance requirements only require this cache
+    # to persist for the duration of one request.
+    # A short cache expiration time mitigates the risk in case someone forgets to
+    # increment the cache_version_key above when making a change.
+    location_params = Rails.cache.fetch(cache_key, expires_in: 1.minute) do
```

Comment:
Will one minute be enough for uploads of up to 1000 samples? it might be a good idea to increase it to ensure the expectation that all samples will be covered in one "request"

Reply:
That's a good point. Right now it wouldn't be enough. I'll set it to 10 minutes which should be enough.